### PR TITLE
SRE-344 Build: Fault Injection tests on el8

### PR DESCRIPTION
**Pull request by JohnMalmberg**
Move the Fault Injection tests from el7 to el8.

Signed-off-by: John E Malmberg <john.e.malmberg@intel.com>

Required-githooks: true

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -770,26 +771,27 @@ pipeline {
...
                     agent {
                         dockerfile {
-                            filename 'utils/docker/Dockerfile.centos.7'
+                            filename 'utils/docker/Dockerfile.el.8'
                             label 'docker_runner'
                             additionalBuildArgs dockerBuildArgs(repo_type: 'stable',
                                                                 parallel_build: true,
-                                                                deps_build: true)
+                                                                deps_build: true) +
+                                                ' --build-arg DAOS_KEEP_SRC=yes'
```
Comment (ashleypittman) [inline Jenkinsfile:786]: The dockerfiles should all be similar in the handling of this, if DAOS_KEEP_SRC is required for el8 but not el7 then it's probably better to update the dockerfile.

File: utils/scripts/install-el8.sh [hunk 2]
```diff
@@ -54,6 +54,7 @@ dnf --nodocs install \
     patchelf \
     python3-devel \
     python3-pip \
+    python3-pyxattr \
```
Comment (ashleypittman) [inline utils/scripts/install-el8.sh:57]: This should be installed through pip as it's listed in https://github.com/daos-stack/daos/blob/master/requirements.txt#L8

**Review by ashleypittman — COMMENTED** [review]
There was something odd I had to do to python in docker on el7 to have it load the python libs correctly, https://github.com/daos-stack/daos/blob/master/utils/docker/Dockerfile.centos.7#L104-L110 is the el7 hack but the rest of the dockerfiles look like https://github.com/daos-stack/daos/blob/master/utils/docker/Dockerfile.el.8#L109-L119

File: Jenkinsfile (hunk 1 above)
Comment (JohnMalmberg) [reply Jenkinsfile:786]: I am not sure that the DAOS_KEEP_SRC is still needed after making the other changes.  I need to test that.

File: utils/scripts/install-el8.sh (hunk 2 above)
Comment (JohnMalmberg) [reply utils/scripts/install-el8.sh:60]: We only want to use Pypi installed modules where we have to in CI.
For internal CI, we should have all Python modules available as RPMs.

Comment (ashleypittman) [reply utils/scripts/install-el8.sh:60]: No we don't, we use Pypi for all python where we can to get newer versions and consistency across distros.

**Review by ashleypittman — APPROVED** [review]
When https://github.com/daos-stack/daos/pull/9513 has landed you can remove all the packaging changes from this PR.

File: utils/scripts/install-el8.sh (hunk 2 above)
Comment (JohnMalmberg) [reply utils/scripts/install-el8.sh:60]: Pypi modules can not be used for package building.   Which means our unit testing with builds using Pypi modules is not going to match our functional testing or what is going to be delivered in packages.

Comment (ashleypittman) [reply utils/scripts/install-el8.sh:60]: Unit test builds are completely different to functional testing.  Functional testing as you say uses rpm packages.

File: utils/scripts/install-el8.sh [hunk 3]
```diff
@@ -45,22 +45,37 @@ dnf --nodocs install \
     Lmod \
     lz4-devel \
     make \
+    meson \
```
Comment (ashleypittman) [inline utils/scripts/install-el8.sh:48]: Can you try this PR with no changes made to this this?

**Review by ashleypittman — CHANGES_REQUESTED** [review]

File: utils/scripts/install-el8.sh (hunk 3 above)
Comment (JohnMalmberg) [reply utils/scripts/install-el8.sh]: Current push has all changes to that file reverted.

File: Jenkinsfile [hunk 4]
```diff
@@ -924,8 +924,8 @@ pipeline {
                         sconsBuild parallel_build: true,
                                    scons_args: 'PREFIX=/opt/daos TARGET_TYPE=release BUILD_TYPE=debug',
                                    build_deps: 'no'
-                        sh label: 'Fault injection testing using NLT',
-                           script: './utils/docker_nlt.sh --class-name centos7.fault-injection fi'
+                        sh (script:'''./utils/docker_nlt.sh --class-name el8.fault-injection fi''',
+                            label: 'Fault injection testing using NLT')
```
Comment (brianjmurrell) [inline Jenkinsfile:928]: We've been following a code style of not using `()` for steps in the `Jenkinsfile` unless necessary (I think only when there are no arguments) and starting `sh` steps with the label argument to help a reader understand the purpose of script that follows.

**Review by brianjmurrell — CHANGES_REQUESTED** [review]

File: Jenkinsfile (hunk 4 above)
Comment (JohnMalmberg) [reply Jenkinsfile:928]: Fixed in next push

**Review by brianjmurrell — DISMISSED** [review]

**Review by jolivier23 — APPROVED** [review]

File: Jenkinsfile [hunk 5]
```diff
@@ -946,7 +946,7 @@ pipeline {
                                                         name: 'Fault injection leaks',
                                                         id: 'NLT_client')]
                             junit testResults: 'nlt-junit.xml'
-                            archiveArtifacts artifacts: 'nlt_logs/centos7.fault-injection/'
+                            archiveArtifacts artifacts: 'nlt_logs/el8.fault-injection/'
```
Comment (ashleypittman) [inline Jenkinsfile:949]: You could probably make this more generic which will help with whatever the next upgrade is.  Alternatively neither this line nor L:928 need the disto in, they both just need to match.
```suggestion
                            archiveArtifacts artifacts: 'nlt_logs/'
```

**Review by ashleypittman — APPROVED** [review]

File: Jenkinsfile (hunk 5 above)
Comment (JohnMalmberg) [reply Jenkinsfile:949]: I would like to leave that for a future PR.

Comment (ashleypittman) [reply Jenkinsfile:949]: Absolutely.

**Pull request merged**